packages/ui/src/components/va-carousel/VaCarousel.vue

```diff
@@ -0,0 +1,257 @@
+import { useCarouselAnimation } from './hooks/useCarouselAnimation'
+import { useStateful, statefulComponentOptions } from '../../mixins/StatefulMixin/cStatefulMixin'
+import VaImage from '../va-image'
+import VaButton from '../va-button'
+
+export default defineComponent({
+  name: 'VaCarousel',
+
+  components: { VaImage, VaButton },
+
+  props: {
+    ...statefulComponentOptions.props,
+
+    modelValue: { type: Number, default: 0 },
+    items: { type: Array as PropType<any[]>, required: true },
+
+    // Animations
+    autoscroll: { type: Boolean, default: false },
+    autoscrollInterval: { type: Number, default: 1000 },
+    autoscrollPauseDuration: { type: Number, default: 2000 },
+    loop: { type: Boolean, default: false },
+    infinite: { type: Boolean, default: false },
+    fadeKeyframe: { type: String, default: 'va-carousel-fade-appear 1s' },
+
+    // Visual
+    arrows: { type: Boolean, default: true },
+    indicators: { type: Boolean, default: true },
+    indicatorTrigger: { type: String as PropType<'click' | 'hover'>, default: 'click' },
+    vertical: { type: Boolean, default: false },
+    height: { type: String, default: '300px' },
+    effect: { type: String as PropType<'fade' | 'transition'>, default: 'transition' },
+  },
+
+  emits: ['update:modelValue'],
+
+  setup (props, { emit }) {
+    const items = toRef(props, 'items')
+    const { valueComputed: currentSlide } = useStateful(props, emit, 0)
+
+    const {
+      goTo, next, prev,
+      doShowNextButton, doShowPrevButton,
+    } = useCarousel(props, currentSlide)
+
+    const { withPause, computedSlidesStyle, slides } = useCarouselAnimation(props, currentSlide)
+
+    return {
+      doShowNextButton,
+      doShowPrevButton,
+      computedSlidesStyle,
+      goTo: withPause(goTo),
+      prev: withPause(prev),
+      next: withPause(next),
+      currentSlide,
+      slides,
+    }
+  },
+})
+</script>
+
+<style lang="scss">
+@import "../../styles/resources";
+@import "./_variables.scss";
+
+@keyframes va-carousel-fade-appear {
+  0% {
+    opacity: 0;
+  }
+
+  100% {
+    opacity: 1;
+  }
+}
+
+.va-carousel {
+  display: flex;
+  width: 100%;
+  height: 100%;
+  background: var(--va-carousel-background);
+  box-shadow: var(--va-carousel-box-shadow);
+  border-radius: var(--va-carousel-border-radius);
+  position: relative;
+  overflow: hidden;
+
+  &__content {
+    flex: 1;
+    width: 100%;
+    white-space: nowrap;
+  }
+
+  &__slides {
+    width: 100%;
+    height: 100%;
+    transition: var(--va-carousel-slides-transition);
+  }
+
+  &__slide {
+    display: inline-flex;
+    justify-content: center;
+    align-items: center;
+    width: 100%;
+    height: 100%;
+    overflow: hidden;
+  }
+
+  &__indicators {
+    width: 100%;
+    position: absolute;
+    bottom: 0;
+    display: flex;
+    justify-content: center;
+    z-index: 2;
+    overflow-x: auto;
+    left: 50%;
+    transform: translateX(-50%);
+    padding: 0 0 16px 0;
```

Comment:
Maybe we can use css-variable instead of 15 or 16px (and below in `left`, `right` and so also)